scio-examples/src/main/scala/com/spotify/scio/examples/complete/game/GameStats.scala

```diff
@@ -27,82 +27,117 @@ import org.apache.beam.sdk.options.StreamingOptions
 import org.apache.beam.sdk.transforms.windowing.{IntervalWindow, TimestampCombiner}
 import org.joda.time.format.DateTimeFormat
 import org.joda.time.{DateTimeZone, Duration, Instant}
+// Example: Calculate game statistics: sum of team's scores, average user session length
 
 object GameStats {
 
+  // The schemas for the BigQuery tables to write output to are defined as annotated case classes
   @BigQueryType.toTable
   case class TeamScoreSums(team: String, total_score: Int,
                            window_start: String, processing_time: String)
-
   @BigQueryType.toTable
   case class AvgSessionLength(mean_duration: Double, window_start: String)
 
   // scalastyle:off method.length
   def main(cmdlineArgs: Array[String]): Unit = {
+    // Create `ScioContext` and `Args`
     val (opts, args) = ScioContext.parseArguments[ExampleOptions](cmdlineArgs)
     val sc = ScioContext(opts)
     sc.optionsAs[StreamingOptions].setStreaming(true)
     val exampleUtils = new ExampleUtils(sc.options)
 
+    // Date formatter for full timestamp
     def fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS")
       .withZone(DateTimeZone.forTimeZone(TimeZone.getTimeZone("PST")))
+    // Duration in minutes for windowing of user and team score sums, defaults to 1 hour
     val fixedWindowDuration = args.int("fixedWindowDuration", 60)
+    // Duration in minutes for length of inactivity after which to start a session, defaults to 5m
     val sessionGap = args.int("sessionGap", 5)
+    // Duration in minutes for windowing of user activities, defaults to 30 min
     val userActivityWindowDuration = args.int("userActivityWindowDuration", 30)
 
+    // Read streaming events from PubSub topic, using ms of events as their ID
     val rawEvents = sc.pubsubTopic(args("topic"), idAttribute = "timestamp_ms")
+      // Parse input as a `GameActionInfo` event
       .flatMap(UserScore.parseEvent)
 
+    // Change each event into a tuple of: user, and that user's score
     val userEvents = rawEvents.map(i => (i.user, i.score))
+    // Window user scores over a fixed length of time
     val userScores = userEvents.withFixedWindows(Duration.standardMinutes(fixedWindowDuration))
+    // Calculate users with scores high enough to be anomalous, to remove from users later
     val spammyUsers = calculateSpammyUsers(userScores).asMapSideInput
 
     rawEvents
+      // Window over a fixed length of time
       .withFixedWindows(Duration.standardMinutes(fixedWindowDuration))
+      // Convert to `SCollectionWithSideInput` to use side input at same time as `SCollection` elt
```

Comment:
`elt` ?

Reply:
element, didn't want to hit line limit. will fix!